graylog2-web-interface/src/views/components/messagelist/MessageFields/hooks/useMessageFavoriteFieldsMutation.ts

```diff
@@ -32,13 +35,25 @@ const useMessageFavoriteFieldsMutation = (streams: Array<Stream>, initialFavorit
       const newFavoriteFieldsByStream = Object.fromEntries(
         streams.map((stream) => [
           stream.id,
-          favoritesToSave.filter((f) => stream.favorite_fields.includes(f) || newAddedFields.includes(f)),
+          favoritesToSave.filter((f) => {
+            const streamFavoriteFields = Array.isArray(stream.favorite_fields) ? stream.favorite_fields : [];
+
+            return streamFavoriteFields.includes(f) || newAddedFields.includes(f);
+          }),
         ]),
       );
 
       setIsLoading(true);
```

Comment:
Can we use `useMutation` here? This way we would get loading states etc. for free.